runtime/lua/vim/lsp/util.lua

```diff
@@ -683,6 +684,52 @@ do
       api.nvim_buf_set_virtual_text(bufnr, diagnostic_ns, line, virt_texts, {})
     end
   end
+  function M.buf_diagnostics_statusline(bufnr, diagnostics)
+    all_buffer_diagnostics_counts[bufnr] = { errors=0, warnings=0, hints=0, info=0 }
+    for _, diagnostic in ipairs(diagnostics) do
+      if diagnostic.severity == 1 then
+        all_buffer_diagnostics_counts[bufnr]['errors'] = all_buffer_diagnostics_counts[bufnr]['errors'] + 1
```

Comment:
I do not see where this is ever _decremented_ ...

Reply:
Ok, yep, maybe caching was not needed for this. thanks!